### PR TITLE
Fix globe terrain holes at poles

### LINKED ISSUE #4984: Mesh is broken near the pole on globe with terrain
There's a hole down though planet if Terrain3D is active on the Globe, which is making some glitchy behavior.

This can be seen in the `heatmap-terrain.html` example.

This is not fixed in:
- #4977 

Arguably easiest to start from this branch
- #4977 

Screenshot 2024-11-04 at 12 29 29

### LINKED ISSUE #4990: Globe with/without terrain can zoom too far out
A globe with terrain will by default allow zooming to negative zoom levels, resulting in some undefined behavior. 

The globe will show artifacts, and then jump back to the minZoom.
There should be a hard limit on how far a user is allowed to zoom out.

https://github.com/user-attachments/assets/7f757c47-5620-4b5a-885b-f2c70d577b6f

Screenshot 2024-11-04 at 22 26 46

Repro:
- Open hybrid-terrain example
- Zoom out

### PR DESCRIPTION
**Pull request by kubapelc**
- Fixes #4984
- Fixes #4990

Also fixes an artifact when globe terrain was zoomed out too much. Brings back the globe/terrain/fill-planet-pole test (see #5194), it should no longer be unstable.

### CODE REVIEW

File: src/render/terrain.ts [hunk 1]
```diff
@@ -108,7 +109,7 @@ export class Terrain {
      * GL Objects for the terrain-mesh
      * The mesh is a regular mesh, which has the advantage that it can be reused for all tiles.
      */
-    _mesh: Mesh;
+    _meshDict: { [key: string]: Mesh } = {};
```
Comment (HarelM) [inline src/render/terrain.ts:112]: `meshCache`?

**Comment by HarelM** [comment]
Does it fix the following issue as well? 
- #4990

**Comment by kubapelc** [comment]
> Does it fix the following issue as well?

It fixes the geometry sticking out of the planet, but not the zoom controls/behaviour.

**Comment by HarelM** [comment]
I'll mark it as fixed as I think this is the main concern in this issue, for the problem of min zoom another issue can be opened if someone cares about it enough...

**Review by HarelM — APPROVED** [review]
Thanks!!

**Comment by kubapelc** [comment]
I've just noticed that the pole mesh is also used on mercator and it doesn't look too good, I'll try to fix that.

**Pull request merged**